components/common/Navbar.tsx

```diff
@@ -1,153 +1,121 @@
-                        transition={{ duration: 0.3, delay: 0.3, ease: "easeIn" }}
-                    >
-                        {
-                            openMobileMenu ? (
-                                <X
-                                    onClick={() => setOpenMobileMenu(false)}
-                                />
-                            ) : (
-                                <Image src={'/assets/menu.svg'}
-                                    alt='menu'
-                                    width={20}
-                                    height={20}
-                                    onClick={() => setOpenMobileMenu(true)}
-                                />
-                            )
-                        }
-                    </motion.div>
-                </div>
-
-                <div className={` flex lg:hidden w-full mx-auto  z-30 transition-all duration-500 overflow-hidden
-    ${openMobileMenu ? "h-auto" : "h-0"}
-    `}>
-                    <ul className='w-[98%] mx-auto bg-white rounded-b-[24px]  flex flex-col items-center justify-center gap-7 py-8 ps-slim leading-[21px] text-[18px]  '>
-                        <li
-                            onClick={() => {
-                                scrollToSection("about")
-                                setOpenMobileMenu(false)
-                            }}
-                        >
-                            About us</li>
-                        <li
-                            onClick={() => {
-                                scrollToSection("features")
-                                setOpenMobileMenu(false)
-                            }}
-                        >
-                            Features</li>
-
-                        <li
-                            onClick={() => {
-                                scrollToSection("join")
-                                setOpenMobileMenu(false)
-                            }}
-                        >
-                            Join</li>
-
-                        <li>
-                            <BeAGenie />
-                        </li>
-
-                    </ul>
-
-                </div>
-            </div>
-        </div >
-
-    )
+// /components/common/Navbar.tsx
+"use client";
+import React, { useState, useEffect, useRef } from 'react';
+import Image from 'next/image';
+import Logo from './Logo';
+import { motion, useAnimation, useInView } from 'framer-motion';
+import { X } from 'lucide-react';
+import JoinNowButton from './JoinNowButton';
+import BeAGenie from './BeAGenie';
+import { scrollToSection } from '@/utils/navgiation';
+import { images } from '@/constants/assets';
+
+interface NavbarProps {
+  menuItems: Array<{ label: string; sectionId: string }>;
+  showJoinButton?: boolean;
+  showBeAGenieButton?: boolean;
 }
 
-export default Navbar
\ No newline at end of file
+const Navbar: React.FC<NavbarProps> = ({ menuItems, showJoinButton = true, showBeAGenieButton = false }) => {
```

Comment:
awesome! thank you for consolidating. i still see another filed named `Navbar.tsx` under `components/GenieComponents/Navbar.tsx`

Can we remove the files that are no longer referenced, just to avoid confusion on which component is still needed.

Reply:
and same for other unused components like `components/GenieComponents/Footer.tsx`